pkg/api/validation/schema.go

```diff
@@ -311,6 +311,9 @@ func (s *SwaggerSchema) validateField(value interface{}, fieldName, fieldType st
 		return s.ValidateObject(value, fieldName, fieldType)
 	}
 	allErrs := []error{}
+	if reflect.TypeOf(value) == nil {
```

Comment:
Hi @kargakis , I have (locally) : 

1. moved the check up to the top of the function.
2. changed the check as you suggested, but it has the same behaviour as before (it panics)

./kubectl.sh --server=http://localhost:8080 create -f ../../test-files/test-rc.yaml
panic: reflect: call of reflect.Value.IsNil on string Value

goroutine 1 [running]:
panic(0x1795040, 0xc82027e520)
	/usr/lib/go-1.6/src/runtime/panic.go:464 +0x3e6
reflect.Value.IsNil(0x14b9240, 0xc8202f1bb0, 0x98, 0xc8202f1bb0)
	/usr/lib/go-1.6/src/reflect/value.go:978 +0xe6
k8s.io/kubernetes/pkg/api/validation.(*SwaggerSchema).validateField(0xc8200ee750, 0x14b9240, 0xc8202f1bb0, 0xc8202f1ba0, 0xa, 0xc8206aa0d8, 0x6, 0xc82025cbd0, 0x0, 0x0, ...)
	/home/angus/work/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/api/validation/schema.go:302 +0xd3
Note the following text from https://golang.org/pkg/reflect/#Value.IsNil
" if v was created by calling ValueOf with an uninitialized interface variable i, i==nil will be true but v.IsNil will panic as v will be the zero Value. "

So I am going to stick with the nil check so we can prevent the panic. I'll update the PR so that the check is at the beginning of the function.